Sources/SwiftToolchain/Toolchain.swift

```diff
@@ -106,6 +106,24 @@ public final class Toolchain {
       return nil
     }
   }
+    
+  public func inferSourcePaths() throws -> [String] {
+    let package = try Package(with: swiftPath, terminal)
+    
+    let targetPaths = package.targets.map { target -> String in
+        guard let path = target.path else {
+            switch target.type {
+            case .regular:
+                return "Sources/\(target.name)"
+            case .test:
+                return "Tests/\(target.name)"
```

Comment:
**suggestion:** not sure we should watch test paths, at least not in this function, which is called from `carton dev`. `carton test` would probably need to call a separate function to watch both sources and tests, but just watching sources here should be enough.

```suggestion
    let targetPaths = package.targets.compactMap { target -> String? in
        guard let path = target.path else {
            switch target.type {
            case .regular:
                return "Sources/\(target.name)"
            case .test:
                return nil
```